test/addons-napi/test_string/test_string.c

```diff
@@ -1,6 +1,46 @@
 #include <node_api.h>
 
-void Copy(napi_env env, napi_callback_info info) {
+void TestLatin1(napi_env env, napi_callback_info info) {
+  napi_status status;
+
+  size_t argc;
+  status = napi_get_cb_args_length(env, info, &argc);
+  if (status != napi_ok) return;
+
+  if (argc < 1) {
+    napi_throw_type_error(env, "Wrong number of arguments");
+    return;
+  }
+
+  napi_value args[1];
+  status = napi_get_cb_args(env, info, args, 1);
+  if (status != napi_ok) return;
+
+  napi_valuetype valuetype;
+  status = napi_typeof(env, args[0], &valuetype);
+  if (status != napi_ok) return;
+
+  if (valuetype != napi_string) {
+    napi_throw_type_error(env, "Wrong type of argments. Expects a string.");
+    return;
+  }
+
+  char buffer[128];
+  int buffer_size = 128;
+
+  status =
+      napi_get_value_string_latin1(env, args[0], buffer, buffer_size, NULL);
+  if (status != napi_ok) return;
+
+  napi_value output;
+  status = napi_create_string_latin1(env, buffer, -1, &output);
```

Comment:
Oh, I think this only works because it is C code, where any implicit conversions between numeric types are allowed. But C++ would be more strict about disallowing the conversion from -1 to `size_t`.